### PR TITLE
Fixed emr-containers update-role-trust-policy for non aws partition

### PR DESCRIPTION
**Pull request by lamricky0**
Fix emr-containers update-role-trust-policy for non aws partition

*Issue #, if available:*

*Description of changes:*

* Updated TRUST_POLICY_STATEMENT_FORMAT with correct AWS partition in arn for emr-containers update-role-trust-policy
* Added unit test to test update-role-trust-policy in a CN partition

*Testing:*

Tested update-role-trust-policy against a EKS cluster in cn-north-1

```
aws emr-containers update-role-trust-policy \
       --cluster-name lamricky-test \
       --namespace namespace \
       --role-name lamrickyJobExecutionRole
```
The following policy was added:

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Service": "ec2.amazonaws.com.cn"
      },
      "Action": "sts:AssumeRole"
    },
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "arn:aws-cn:iam::065596917820:oidc-provider/oidc.eks.cn-north-1.amazonaws.com.cn/id/03844B1E626AF1801BDABC572F744B3E"
      },
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Condition": {
        "StringLike": {
          "oidc.eks.cn-north-1.amazonaws.com.cn/id/03844B1E626AF1801BDABC572F744B3E:sub": "system:serviceaccount:namespace:emr-containers-sa-*-*-065596917820-oz977czyn9caw6ow33q8eppc4tg1kr4kekspx"
        }
      }
    }
  ]
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: awscli/customizations/emrcontainers/constants.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 TRUST_POLICY_STATEMENT_FORMAT = '{ \
     "Effect": "Allow", \
     "Principal": { \
-        "Federated": "arn:aws:iam::%(AWS_ACCOUNT_ID)s:oidc-provider/' \
+        "Federated": "arn:%(POLICY_ARN)s:iam::%(AWS_ACCOUNT_ID)s:oidc-provider/' \
```
[the remark this answers is not in the document]
Comment (lamricky0) [reply awscli/customizations/emrcontainers/constants.py]: Addressed in new commit.

File: tests/unit/customizations/emrcontainers/test_update_assume_role_policy.py [hunk 2]
```diff
@@ -229,6 +250,29 @@ def test_trust_policy_exists(self, get_account_id_patch,
...
+    @mock.patch('awscli.customizations.emrcontainers.'
+                'iam.IAM.get_assume_role_policy')
+    @mock.patch('awscli.customizations.emrcontainers.'
+                'eks.EKS.get_oidc_issuer_id')
+    @mock.patch('awscli.customizations.emrcontainers.'
+                'eks.EKS.get_account_id')
+    @mock.patch('awscli.customizations.emrcontainers.'
+                 'update_role_trust_policy.get_region')
+    def test_trust_policy_does_not_exist_in_cn(self, get_region_patch,
+                                         get_account_id_patch,
```
[the remark this answers is not in the document]
Comment (lamricky0) [reply tests/unit/customizations/emrcontainers/test_update_assume_role_policy.py]: Addressed in new commit.

File: tests/unit/customizations/emrcontainers/test_update_assume_role_policy.py [hunk 3]
```diff
@@ -229,6 +250,29 @@ def test_trust_policy_exists(self, get_account_id_patch,
...
+    # Use case: Expected trust policy does not exist in cn-north-1
+    # Expected results: Operation is performed by client in cn-north-1
+    # to update the trust policy in expected format
+    @mock.patch('awscli.customizations.emrcontainers.'
+                'iam.IAM.get_assume_role_policy')
+    @mock.patch('awscli.customizations.emrcontainers.'
+                'eks.EKS.get_oidc_issuer_id')
+    @mock.patch('awscli.customizations.emrcontainers.'
+                'eks.EKS.get_account_id')
+    @mock.patch('awscli.customizations.emrcontainers.'
```
[the remark this answers is not in the document]
Comment (lamricky0) [reply tests/unit/customizations/emrcontainers/test_update_assume_role_policy.py]: Addressed in new commit.

File: awscli/customizations/emrcontainers/constants.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 TRUST_POLICY_STATEMENT_FORMAT = '{ \
```
[the remark this answers is not in the document]
Comment (lamricky0) [reply awscli/customizations/emrcontainers/constants.py:17]: Addressed in new commit.

File: tests/unit/customizations/emrcontainers/test_update_assume_role_policy.py [hunk 5]
```diff
@@ -67,8 +79,12 @@ def setUp(self):
         self.expected_policy_document.get("Statement").append(
             json.loads(self.expected_statement))
 
+        self.expected_policy_document_cn = copy.deepcopy(self.policy_document)
+        self.expected_policy_document_cn.get("Statement").append(
+            json.loads(self.expected_statement_cn))
+
     # Assert the call to update trust policy of the role
-    def assert_trust_policy_updated(self, cmd_output):
+    def assert_trust_policy_updated(self, cmd_output, cn = False):
```
[the remark this answers is not in the document]
Comment (lamricky0) [reply tests/unit/customizations/emrcontainers/test_update_assume_role_policy.py]: Addressed in new commit.

**Review by kyleknap — APPROVED** [review]
Looks good to me! :ship:

**Pull request merged**